crates/ide-assists/src/handlers/generate_getter.rs

```diff
@@ -72,88 +75,263 @@ pub(crate) fn generate_getter_mut(acc: &mut Assists, ctx: &AssistContext<'_>) ->
     generate_getter_impl(acc, ctx, true)
 }
 
+#[derive(Clone, Debug)]
+struct RecordFieldInfo {
+    field_name: syntax::ast::Name,
+    field_ty: syntax::ast::Type,
+    fn_name: String,
+    target: TextRange,
+}
+
+struct GetterInfo {
+    impl_def: Option<ast::Impl>,
+    strukt: ast::Struct,
+    mutable: bool,
+}
+
 pub(crate) fn generate_getter_impl(
     acc: &mut Assists,
     ctx: &AssistContext<'_>,
     mutable: bool,
 ) -> Option<()> {
-    let strukt = ctx.find_node_at_offset::<ast::Struct>()?;
-    let field = ctx.find_node_at_offset::<ast::RecordField>()?;
+    // This if condition denotes two modes this assist can work in:
+    // - First is acting upon selection of record fields
+    // - Next is acting upon a single record field
+    //
+    // This is the only part where implementation diverges a bit,
+    // subsequent code is generic for both of these modes
+
+    let (strukt, info_of_record_fields, fn_names) = if !ctx.has_empty_selection() {
+        // Selection Mode
+        let node = ctx.covering_element();
+
+        let node = match node {
+            syntax::NodeOrToken::Node(n) => n,
+            syntax::NodeOrToken::Token(t) => t.parent()?,
+        };
+
+        let mut parent_struct: Option<ast::Struct> = None;
+        if let Some(struct_node) = node.ancestors().find(|it| ast::Struct::can_cast(it.kind())) {
+            parent_struct = ast::Struct::cast(struct_node);
+        }
+
+        // If we don't have a parent struct, something is wrong, OPERATION ABORT 🚨
+        if parent_struct.is_none() {
+            return None;
+        }
+
+        let (info_of_record_fields, field_names) =
+            extract_and_parse_record_fields(&node, ctx.selection_trimmed(), mutable)?;
```

Comment:
```suggestion
            extract_and_parse_record_fields(&parent_struct, ctx.selection_trimmed(), mutable)?;
```
No need to use the untyped node here